TESTS/netsocket/tcp/tcpsocket_echotest.cpp

```diff
@@ -83,33 +92,31 @@ void TCPSOCKET_ECHOTEST()
     TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock.close());
 }
 
-void tcpsocket_echotest_nonblock_receiver(void *receive_bytes)
+void tcpsocket_echotest_nonblock_receive()
 {
-    int bytes2recv = *(int *)receive_bytes;
-    int recvd;
-    while (bytes2recv) {
-        recvd = sock.recv(&(tcp_global::rx_buffer[*(int *)receive_bytes - bytes2recv]), bytes2recv);
-        if (recvd == NSAPI_ERROR_WOULD_BLOCK) {
-            if (tc_exec_time.read() >= time_allotted) {
-                TEST_FAIL();
-                break;
-            }
-            wait(1);
-            continue;
-        } else if (recvd < 0) {
-            TEST_FAIL();
-            break;
+    int recvd = sock.recv(&(tcp_global::rx_buffer[bytes2recv_total - bytes2recv]), bytes2recv);
+    if (recvd == NSAPI_ERROR_WOULD_BLOCK) {
+        if (tc_exec_time.read() >= time_allotted) {
+            receive_error = true;
```

Comment:
tx_sem isn't released here. Sender might end up waiting forever as the timeout was removed.

Reply:
We don't really want to release the semaphore in case of WOULD_BLOCK - we want to wait until the data arrives. Therefore, I added the timeout back, but fixed it to the correct units.
I hope this is fine? :)